src/Components/useVerifyInput.js

```diff
@@ -0,0 +1,52 @@
+import { debounce } from 'lodash';
+import { useState, useEffect, useRef, useCallback } from 'react';
+
+export const useVerifyInput = (input, executeMutation) => {
+  const [verifiedStatus, setVerifiedStatus] = useState('');
+  const isMounted = useRef(false);
+  const initialRender = useRef(true);
+
+  const debouncedExecuteMutation = useCallback(
+    debounce(async () => {
+      try {
+        await executeMutation();
+        if (isMounted.current) {
+          setVerifiedStatus('success');
+        }
+      } catch (error) {
+        console.error('Error executing mutation:', error);
+        if (isMounted.current) {
+          setVerifiedStatus('error');
+        }
+      }
+    }, 1000), // 1 second delay
+    [executeMutation]
+  );
+
+  useEffect(() => {
+    isMounted.current = true;
```

Comment:
For my understanding, why we need these isMounted and initialRender refs?

Reply:
The component can rerender off a few inputs from a user. When because of that, sometimes the hook was firing off infinitely cause some dependency was updated.
I made initialRender so to stop the hook from running on page load(the user hasnt typed in anything yet)
I made isMounted because in the async function, the request sometimes completed when the component unmounted. When you try to update the components start after it was unmounted, react will throw a memory leak warning. IsMounted is more just to be safe.

For how isMounted works, the useEffect sets isMounted.current to true when the component initially mounts (we only want to run when the component is mounted). When the component unmounts, the useEffect sets it to false. This makes ongoing async tasks that would cause any state updates be aborted because the component isnt technically in the DOM anymore. Im not entirely sure why the component unmounts here, but this was to circumvent that. Since this works with debounce, theres a second where the verifiedStatus is in 'loading' state. Im thinking this is when the component 'unmounts'.